### PR TITLE
Add a workflow to generate the header file from the SMILES templates file

### PR DESCRIPTION
**Pull request by ricrogz**
This new workflow is triggered when there are updates to the `templates.smi` file on the main branch. When that happens, this workflow triggers, running the following steps:
1. Set up the work environment.
2. Run the checks on the templates (just in case).
3. Run the script that generates the header file.
4. Check that the header file can be compiled into a dummy program.
5. Create/update a gallery file with the current templates.
6. Commit the new header file into the `main` branch of this repository.

### CODE REVIEW

**Comment by rachelnwalker** [comment]
This is great Ricardo! Am I understanding correctly that gallery.md will be created and included in this repo? Do you have an example of what that currently looks like?

**Comment by ricrogz** [comment]
> This is great Ricardo! Am I understanding correctly that gallery.md will be created and included in this repo? Do you have an example of what that currently looks like?

There's an example in this branch: https://github.com/ricrogz/molecular_templates/tree/header_generator_example. The gallery file in that branch is here: https://github.com/ricrogz/molecular_templates/blob/header_generator_example/gallery.md

(Note that the last template, #73, is just the dummy I used to trigger the workflow -- it's basically a clone of #72 in which all C atoms have been replaced with N).

File: checks/test_program.cpp [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#include <iostream>
```
Comment (d-b-w) [inline checks/test_program.cpp:1]: A really nice touch to make sure that the header compiles and that the data is printable.

**Review by d-b-w — APPROVED** [review]
This looks great! Ship it

**Pull request merged**